File: backend/src/main/java/com/krishimitra/farm/Farm.java
package com.krishimitra.farm;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "farms")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Farm {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    
    @Column(name = "farmer_id")
    private UUID farmerId;
    
    private String name;
    
    @Column(precision = 10, scale = 6)
    private BigDecimal latitude;
    
    @Column(precision = 10, scale = 6)
    private BigDecimal longitude;
    
    @Column(name = "area_acres", precision = 6, scale = 2)
    private BigDecimal areaAcres;
    
    @Column(name = "soil_type")
    private String soilType;
    
    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
}

File: mobileV2/app/src/main/java/com/krishimitra/mobilev2/data/model/Farm.kt
package com.krishimitra.mobilev2.data.model

/**
 * Farm entity — represents a farmer's farm/land parcel.
 */
data class Farm(
    val id: String = "",
    val farmerId: String,
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val areaAcres: Double,
    val soilType: String
)

/**
 * Farm API response
 */
data class FarmResponse(
    val farm_id: String,
    val farmer_id: String,
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val area_acres: Double,
    val soil_type: String
) {
    fun toFarm(): Farm = Farm(
        id = farm_id,
        farmerId = farmer_id,
        name = name,
        latitude = latitude,
        longitude = longitude,
        areaAcres = area_acres,
        soilType = soil_type
    )
}